data-8/jupyterhub-k8s
# Refactor, generalize and upstream various changes

Summary
-------

I've merged generalized changes from at the fix_tini branch here. The changes from @derrickmar's 'hardcoded2' branch of kubespawner have also been merged upstream, and related changes have been made here. The ideal end goal is for this set of helm charts to be able to do 'install a JupyterHub with a specific config on a kubernetes cluster' - it isn't there yet, but getting towards it.

I've tried to attribute the work where I can, but am sure I've missed some! Let me know what else I need to add and I'll happily do so!

How to test
-------

1. Add appropriate values to `data8-jhub/values.yaml`.
2. Run `helm install --debug data8-jhub` from the base git directory
3. Test if it all works!

How to push new container images to gcr.io:
----------
This patch uses gcr.io as the docker image registry, rather than dockerhub. gcr.io is hosted by google (it is the Google Container Registry), in the same datacenter as our GKE install - this makes it a lot faster. 

To build the hub container:
1. cd hub
2. docker build -t gcr.io/data-8/data8-hub .
3. gcloud docker -- push gcr.io/data-8/data8-hub

To build the user container:
1. cd user
2. docker build -t gcr.io/data-8/data8-singleuser
3. gcloud docker -- push gcr.io/data-8/data8-singleuser

Note that the singleuser container is a bit fatter after the refactor - needs to be trimmed down later.
